solidpy/Grain.py

```diff
@@ -13,24 +13,33 @@ def __init__(
         outer_radius,
         initial_inner_radius,
         initial_height=None,
+        mass=None,
         geometry="tubular",
     ):
 
         self.outer_radius = outer_radius
         self.initial_inner_radius = initial_inner_radius
         self.inner_radius = initial_inner_radius
+        self.mass = mass
         self.evaluate_grain_initial_height(initial_height)
         self.height = self.initial_height
         self.geometry = geometry
         self.evaluate_grain_geometry()
         self.evaluate_grain_volume()
+        self.density = self.evaluate_grain_density()
 
     def evaluate_grain_initial_height(self, initial_height):
         if initial_height is None:
             self.initial_height = 3 * self.outer_radius + self.inner_radius
         else:
             self.initial_height = initial_height
 
+    def evaluate_grain_density(self):
+        if self.mass is not None:
+            density = self.mass / self.volume
+            return density
+        return None
```

Comment:
While this is a nice feature, I think it's not very clear here that it overrides the defined density inside the `Propellant` class (of which the `Grain` class is not aware). Some documentation explaining the inner functioning would be welcome.

Also, it's pretty unusual for grain mass to be available until a few days before firing (specially for sugar propellants, due to moisture), at which point running simulations wouldn't be very useful for motor design. Perhaps we can somehow find a way to bundle this together with the other "empirical" features, as suggested above?